### PR TITLE
fixes #13854 : Reduce message log height by 1 to show full status height in wider siderbar mode

### LINKED ISSUE #13854: Vehicle speed indicator missing in "wider" sidebar mode
Seems to be replaced by the new moon phase indicator.

### PR DESCRIPTION
**Pull request by remyroy**
Related to #13854.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -452,7 +452,7 @@ void game::init_ui()
         messX = MINIMAP_WIDTH;
         messY = 0;
         messW = sidebarWidth - messX;
-        messH = TERRAIN_WINDOW_TERM_HEIGHT - 5; // 1 for w_location + 4 for w_stat, w_messages starts at 0
+        messH = TERRAIN_WINDOW_TERM_HEIGHT - 6; // 1 for w_location + 4 for w_stat, w_messages starts at 0
```
Comment (drbig) [inline src/game.cpp]: Update the comment too please.

**Comment by remyroy** [comment]
Good catch on the comment. By reading it again, I found a bug in the fix as well. Everything should be fine now.

**Comment by remyroy** [comment]
I've been thinking about an alternative to this fix. So instead of having a comment and a fix value for:

```c++
messH = TERRAIN_WINDOW_TERM_HEIGHT - 6;
```

what if we define locH and statH first than we use:

```c++
messH = TERRAIN_WINDOW_TERM_HEIGHT - locH - statH;
```

This way we don't need to define why we use the fixed 6 and if someone changes locH again, he won't need to change that fixed value as well. The con is that we will have to split the defining values for the w_information window meaning that they will not be logically together in the source.

A hard call to me. What do you think?

**Comment by BevapDin** [comment]
You can probably shuffle the initialization of those variables a bit around to avoid the dependencies in the comments. Separating the position values from the size values shouldn't be much of a problem. The code above (for the narrow sidebar) already looks like that.

Btw. wouldn't you need to change `statH` as well? It's still 4, so the status window will still be 4 lines height.

Btw. if you get rid of the implicit dependencies, please also remove the redundant comments about it. Nobody needs comments that only repeat the code.

**Comment by remyroy** [comment]
I moved the `locH` and `statH` initialization before the `messH` one.

statH should be 4. The reason why `w_status` was not fully showing in #13854 is because `statY` got push too far down with `messH` being too large.

**Pull request merged**